darwin/future/meta/objects/dataset.py

```diff
@@ -1,10 +1,17 @@
-from typing import List, Optional, Tuple, Union
+from __future__ import annotations
```

Comment:
With this, I tend to just quote the returnable I need it for, but this is valid too, especially with our current support matrix.